### PR TITLE
fix #303: gdlib.pc: use Requires instead of Libs

### LINKED ISSUE #303: gdlib.pc: use Requires instead of Libs when possible
gdlib.pc adds a list of libraries to add the linker command lines, which seems a bit 'strange'.

Wouldn't it be more natural to add the required dependencies to the .pc Requires: field?

### PR DESCRIPTION
**Pull request by willson-chen**
gdlib.pc: use Requires instead of Libs #303

### CODE REVIEW

**Comment by vapier** [comment]
can you rebase ?

**Comment by willson-chen** [comment]
> can you rebase ?

done

**Comment by willson-chen** [comment]
@vapier  sorry. misunderstood that. it is really done this time. :)

File: configure.ac [hunk 1]
```diff
@@ -161,6 +165,11 @@ m4_define([GD_LIB_CHECK], [dnl
       $1][_CFLAGS="$gd_lib_cflags $][$1][_CFLAGS"
       AS_VAR_APPEND([LIBS], [" $][$1][_LIBS"])
       AS_VAR_APPEND([CPPFLAGS], [" $][$1][_CFLAGS"])
+      if test "$gd_require_pkg_name" = ""; then
```
Comment (vapier) [inline configure.ac]: use `-z`

File: configure.ac [hunk 2]
```diff
@@ -180,7 +189,7 @@ dnl $4 - pkg-config module to look for
 dnl $5 - fallback test for the feature
 m4_define([GD_LIB_PKG_CHECK], [dnl
   GD_LIB_CHECK([$1], [$2], [$3], [dnl
-    PKG_CHECK_MODULES([$1], [$4], [gd_found_lib=yes], [$5])
+    PKG_CHECK_MODULES([$1], [$4], [gd_found_lib=yes gd_require_pkg_name=["$4"]], [$5])
```
Comment (vapier) [inline configure.ac]: does this need the `[]` around `"$4"` ?  i don't think it does.

File: configure.ac (hunk 1 above)
Comment (willson-chen) [reply configure.ac]: done

File: configure.ac (hunk 2 above)
Comment (willson-chen) [reply configure.ac]: done

**Comment by willson-chen** [comment]
Hi @cmb69 . Could you please spare some time to review this PR? :)

**Pull request merged**